src/xtgeo/surface/regular_surface.py

```diff
@@ -2723,9 +2739,11 @@ def slice_cube_window(
                 PendingDeprecationWarning,
             )
 
+        scube = surface_from_cube(cube, 0)
```

Comment:
This is a note-worthy change: this surface is made at the top-level function call and then passed back to reduce circular dependencies in the `slice_cube_window` files. This should be much simpler if and when the different versions of algorithms are removed and cohered down to one file (or just this function).